tests/type_check/mod.rs

```diff
@@ -48,3 +48,59 @@ macro_rules! test_type_with_placeholder {
         }
     };
 }
+
+/// Create a test with `$name` that type checks a file with `$name`.pj
+/// in the same directory against the `$pattern`. It does this once for
+/// each int binary operator that will replace the `$placeholder` in the file.
+#[macro_export]
+macro_rules! test_type_with_all_int_binops {
```

Comment:
I think that `test_type_for_all_integer_binops` sounds better. Same for the other macros.